provisioning/utils.go

```diff
@@ -47,11 +47,19 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
 	return hostIP, err
 }
 
-func GetIronicIP(client kubernetes.Interface, targetNamespace string, config *metal3iov1alpha1.ProvisioningSpec) (string, error) {
-	// if UseIronicProxy(config) {
-	// TODO: API VIP ???
-	// } else
-	if config.ProvisioningNetwork != metal3iov1alpha1.ProvisioningNetworkDisabled && !config.VirtualMediaViaExternalNetwork {
+func GetIronicIP(client kubernetes.Interface, targetNamespace string, config *metal3iov1alpha1.ProvisioningSpec, info *ProvisioningInfo) (string, error) {
+	if UseIronicProxy(config) {
+		ctx := context.Background()
+		infra, err := info.OSClient.ConfigV1().Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+		if err != nil {
+			return "", err
```

Comment:
Let's use errors.Wrap to provide a meaningful context

Reply:
Thank you, done.